planner:
  mode: "VisRRT" # "VisRRT" or "VisPRM"
  use_visual_ik: true
  
  # Planning Group & Links
  group_name: "manipulator"
  ee_link_name: "ur_arm_tool0"
  # ee_link_name: "flashlight"

  # # General Settings
  resolution: 0.01
  visibility_threshold: 0.8

  # Visibility tool Parameters
  visibility_tool_params:
    beam_length: 1.0
    beam_angle: 0.21

  # Post-Processing
  enable_shortcutting: true

  # RRT Specific Parameters
  rrt:
    goal_bias: 0.01
    max_extension: 3.0
    max_iterations: 1000

  # # PRM Specific Parameters
  # prm:
  #   num_neighbors: 10
  #   num_samples: 1000
  #   # 0 = LINEAR, 1 = BINARY_SEARCH
  #   edge_validation_method: 1

  # Execution Settings (Used by Client)
  execution:
    velocity_scaling: 0.1
    acceleration_scaling: 0.1
